### PR TITLE
[camera] Ignore body_might_complete_normally_catch_error violation

### PR DESCRIPTION
**Pull request by srawlins**
Ignore body_might_complete_normally_catch_error violation

No version change: Ignore directives do not affect plugin clients.

Related to https://github.com/flutter/flutter/issues/105750

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart [hunk 1]
```diff
@@ -130,7 +130,11 @@ class MethodChannelCamera extends CameraPlatform {
         'cameraId': cameraId,
         'imageFormatGroup': imageFormatGroup.name(),
       },
-    ).catchError(
+    )
+        // TODO(srawlins): Fix this static issue,
+        // https://github.com/flutter/flutter/issues/105750.
```
[the remark this answers is not in the document]
Comment (srawlins) [reply packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart]: Done

**Comment by srawlins** [comment]
Done

**Comment by stuartmorgan** [comment]
I've added an explanation to the PR description for why there's no version change, which will make `publishable` pass. You'll need to push an update to this PR to get it to re-run though due to some caching of PR descriptions that happens somewhere in the CI.

File: packages/camera/camera_platform_interface/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
 
 * Ignores unnecessary import warnings in preparation for [upcoming Flutter changes](https://github.com/flutter/flutter/pull/104231).
+* Ignores missing return warnings in preparation for [upcoming analysis changse](https://github.com/flutter/flutter/issues/105750).
```
Comment (bparrishMines) [inline packages/camera/camera_platform_interface/CHANGELOG.md:4]: nit:

```suggestion
* Ignores missing return warnings in preparation for [upcoming analysis changes](https://github.com/flutter/flutter/issues/105750).
```

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**